optimade/validator/validator.py

```diff
@@ -483,32 +489,36 @@ def _construct_queries_for_property(
                 f"Chosen entry of endpoint '/{endp}' had unexpected type '{chosen_entry['type']}'."
             )
 
-        if prop == "id":
-            prop_type = DataType.STRING
+        if prop_type is None:
+            prop_type = CONF.entry_schemas[endp].get(prop, {}).get("type")
```

Comment:
To avoid codecov complaining:
```suggestion
        prop_type = CONF.entry_schemas[endp].get(prop, {}).get("type") if prop_type is None else prop_type
```

Reply:
I think yours is technically more characters :P

Reply:
Indeed, this is all about codecov not complaining it doesn't reach the line within the `if`-block.